server/app/durablejobs/jobs/CopyFileKeyForMultipleFileUpload.java

```diff
@@ -0,0 +1,102 @@
+ * uploads.
+ *
+ * <p>Looks at applicant data for any node containing "file_key" and creates a sibling node
+ * "file_keys" with the same data.
+ */
+public final class CopyFileKeyForMultipleFileUpload extends DurableJob {
+  private static final String FILE_KEY_PROPERTY = Scalar.FILE_KEY.name().toLowerCase(Locale.ROOT);
+  private static final String FILE_KEY_LIST_PROPERTY =
+      Scalar.FILE_KEY_LIST.name().toLowerCase(Locale.ROOT);
+
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(CopyFileKeyForMultipleFileUpload.class);
+
+  private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+  private final PersistedDurableJobModel persistedDurableJobModel;
+  private final AccountRepository accountRepository;
+  private final ApplicationRepository applicationRepository;
+
+  public CopyFileKeyForMultipleFileUpload(
+      PersistedDurableJobModel persistedDurableJobModel,
+      AccountRepository accountRepository,
+      ApplicationRepository applicationRepository) {
+    this.persistedDurableJobModel = persistedDurableJobModel;
+    this.accountRepository = accountRepository;
+    this.applicationRepository = applicationRepository;
+  }
+
+  @Override
+  public PersistedDurableJobModel getPersistedDurableJob() {
+    return persistedDurableJobModel;
+  }
+
+  @Override
+  public void run() {
+    LOGGER.atInfo().log("Copying file keys for applicants.");
+
+    try (QueryIterator<ApplicantModel> applicants = accountRepository.streamAllApplicants()) {
+      while (applicants.hasNext()) {
+        try {
+          ApplicantModel applicant = applicants.next();
+          ApplicantData migratedData = migrateApplicantData(applicant.getApplicantData());
+          applicant.setApplicantData(migratedData);
+          applicant.save();
+        } catch (Exception e) {
+          LOGGER.error(e.getMessage(), e);
+        }
+      }
+    }
+
+    LOGGER.atInfo().log("Copying file keys for applications.");
+
+    try (QueryIterator<ApplicationModel> applications =
+        applicationRepository.streamAllApplications()) {
+      while (applications.hasNext()) {
+        try {
+          ApplicationModel application = applications.next();
+          application.setApplicantData(migrateApplicantData(application.getApplicantData()));
+          application.save();
+        } catch (Exception e) {
+          LOGGER.error(e.getMessage(), e);
+        }
+      }
+    }
+
+    LOGGER.atInfo().log("Finished copying file keys.");
+  }
+
+  @VisibleForTesting
+  ApplicantData migrateApplicantData(ApplicantData applicantData) throws Exception {
+    JsonNode rootJsonNode = OBJECT_MAPPER.readTree(applicantData.asJsonString());
+
+    for (JsonNode node : rootJsonNode.findParents(FILE_KEY_PROPERTY)) {
+      if (!node.has(FILE_KEY_LIST_PROPERTY)) {
+        JsonNode fileKeyNode = node.get(FILE_KEY_PROPERTY);
+        ((ObjectNode) node).putArray(FILE_KEY_LIST_PROPERTY).add(fileKeyNode);
```

Comment:
(question) Is this adding the value to the array or a node of `file_key`?

Reply:
It's really weird. First putArray is adding an empty array node to our parent node (the node containing file_key). This is the "file_key_list" node. Then we're adding the value node from "file_key" to that array (so now file_key_list is an array with one entry, the value of "file_key"). 

This seemed to be the right way to do it, I agree it's confusing and counterintuitive.

Reply:
Yeah that is weird. Can you run your job locally and check the json in the database to be double sure it's right.

Reply:
Yep, it modifies it correctly.

Reply:
Could you also add a comment into the file to note the behavior?